src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ComInterop/VarEnumSelector.cs

```diff
@@ -97,8 +97,8 @@ internal static Type GetManagedMarshalType(VarEnum varEnum)
                 { VarEnum.VT_UI2,       typeof(ushort) },
                 { VarEnum.VT_UI4,       typeof(uint) },
                 { VarEnum.VT_UI8,       typeof(ulong) },
-                { VarEnum.VT_INT,       typeof(IntPtr) },
-                { VarEnum.VT_UINT,      typeof(UIntPtr) },
+                { VarEnum.VT_INT,       typeof(int) },
+                { VarEnum.VT_UINT,      typeof(uint) },
                 { VarEnum.VT_BOOL,      typeof(bool) },
```

Comment:
There are also other parts of the runtime that consider `VT_INT`/`VT_UINT` to be `IntPtr`/`UIntPtr`: https://github.com/dotnet/runtime/blob/a1c19db3ff60d42b690db5b3d2ba866c99d41ffd/src/libraries/Common/src/System/Runtime/InteropServices/Variant.cs#L254-L255

I don't remember the specifics, but I recall @AaronRobinsonMSFT looking at something related to this when dealing with tests: https://github.com/dotnet/runtime/blob/a1c19db3ff60d42b690db5b3d2ba866c99d41ffd/src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs#L39-L41

Reply:
There are other places where `VT_INT` and `VT_UINT` are treated as `int` and `uint`. See [`Variant.cs`](https://github.com/dotnet/runtime/blob/master/src/libraries/Common/src/System/Runtime/InteropServices/Variant.cs#L73-L74)

```c#
        private struct UnionTypes
        {
            ...
            [FieldOffset(0)] public int _int;
            [FieldOffset(0)] public uint _uint;
            ...
        }
```

Reply:
Okay - looked through this some more. We should fix the `VT_INT`/`VT_UINT` -> `int`/`uint` mapping (this map). We should also update:
https://github.com/dotnet/runtime/blob/a1c19db3ff60d42b690db5b3d2ba866c99d41ffd/src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ComInterop/Variant.Extended.cs#L70-L82

But the conversion from `IntPtr`/`UIntPtr` -> `VT_PTR` would be problematic (the other change in this file), since we consider `VT_PTR` as supported throughout the runtime and specifically handle `IntPtr`/`UIntPtr` to be `VT_INT`/`VT_UINT`. I think that part needs to stay as-is.

Reply:
I really don't have enough expertise to comment on this, so I trust what you guys decide on after you thoroughly look into it :)

Reply:
We should be good with what I described:
- Keep the fix for mapping `VT_INT`/`VT_UINT` -> `int`/`uint` in `CreateComToManagedPrimitiveTypes`
- Update `SetAsByRefInt` / `SetAsByrefUint` to take `int`/`uint` instead of `IntPtr`/`UIntPtr`
- Undo the change in `TryGetPrimitiveComType`